scopes/core.md

```diff
@@ -0,0 +1,176 @@
+# Core scope type definitions
+
+The following core scope types are available:
+
+- network scope
+- health scope
+
+## Network scope
+
+The network scope groups components together and links them to a network or SDN. The network itself must be defined and operated by the infrastructure.
+
+The network scope may also be queried by traffic management traits to determine discoverability boundaries for a service mesh or API boundaries for API gateways.
+
+If no network scope is assigned, the platform MUST join the application to a default network. Within that default network, all components in the application configuration MUST be able to communicate to each other, and health probes MUST be able to contact any components who define health checking rules. The network joined, however, is platform-dependent. For example, cluster-based environments (such as Kubernetes) declare cluster-wide networks. Conversely, a truly serverless implementation may join the components to a network that only includes the components and the health checking probes.
+
+### Schematic
+
+| Attribute | Type | Required | Default Value | Description |
+|-----------|------|----------|---------------|-------------|
+| `networkId` | `string` | Y | | The id of the network, e.g. vpc-id, VNet name. |
+| `subnetIds` | `array` | Y | | A comma separated list of IDs of the subnets within the network. For example, 'vsw-123' or 'vsw-123','vsw-456'. There could be more than one subnet because there is a limit in the number of IPs in a subnet. If IPs are taken up, operators need to add another subnet into this network. |
+| `internetGatewayType` | `string` | Y | | The type of the gateway, options are 'public', 'nat'. Empty string means no gateway. |
+
+### Spec
+
+The network scope is defined by the following spec:
+
+```yaml
+apiVersion: core.oam.dev/v1alpha1
+kind: ScopeDefinition
+metadata:
+  name: networkscope.core.oam.dev
+spec:
+  allowComponentOverlap: true
+  definitionRef:
+     name: networkscope.core.oam.dev
+```
+
+OAM runtime should store the schema with a version, the schema SHOULD be able to validate. For example using JSON Schema like this:
+
```

Comment:
Is this just an example, or is it meant to be the actual definition of a core network scope?

Reply:
I'm not sure OAM should be defining the schema of these scopes at this point.

I know this was already part of the spec and thus changing it could be out of scope (hurr) for this PR, but I fear that this schema could be a little speculative and naive. Do we know of any implementations that use scopes in the wild? I recall Rudr does not support them, and we've found representing concepts like VPC networks in a portable, abstract fashion to be a hard thing(tm) in Crossplane.